client/src/app/components/stardust/history/TransactionHistory.tsx

```diff
@@ -66,9 +67,17 @@ const TransactionHistory: React.FC<TransactionHistoryProps & AsyncProps> = (
 
             tangleService().transactionHistory(request)
                 .then((response: ITransactionHistoryResponse | undefined) => {
-                    if (response?.items && isMounted) {
-                        setHistory([...history, ...response.items]);
-                        setCursor(response.cursor);
+                    const items = response?.items ?? [];
+                    if (items.length > 0 && isMounted) {
+                        setHistory([...history, ...items]);
+                        setCursor(response?.cursor);
+                    } else {
+                        onAsyncStatusChange(PromiseStatus.DONE);
+                    }
+                })
+                .finally(() => {
+                    if (setTransactionCount) {
+                        setTransactionCount(history.length);
```

Comment:
I see no point in having the count as we load additional pages of data on-demand. This was just show the counter of currently loaded entries.

Reply:
This fixes the transaction tab infinite loading issue because it triggers a re-render on count update. Currently transaction loader will stop loading if any of the other tabs trigger re-rendering by updating their count.